cloud/aws/elasticache/redis/monitors-redis.tf

```diff
@@ -0,0 +1,119 @@
+		notifications         = coalescelist(var.cpu_high_notifications_warning, var.cpu_high_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "replication_lag" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS ElastiCache redis replication lag"
+
+	program_text = <<-EOF
+		signal = data('ReplicationLag', filter=filter('namespace', 'AWS/ElastiCache') and filter('stat', 'upper') and filter('CacheNodeId', '*') and ${module.filter-tags.filter_custom})${var.replication_lag_aggregation_function}.${var.replication_lag_transformation_function}(over='${var.replication_lag_transformation_window}')
+		detect(when(signal > ${var.replication_lag_threshold_critical})).publish('CRIT')
+		detect(when(signal > ${var.replication_lag_threshold_warning})).publish('WARN')
+	EOF
+
+	rule {
+		description           = "is too high > ${var.replication_lag_threshold_critical}"
+		severity              = "Critical"
+		detect_label          = "CRIT"
+		disabled              = coalesce(var.replication_lag_disabled_critical, var.replication_lag_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.replication_lag_notifications_critical, var.replication_lag_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+	rule {
+		description           = "is too high > ${var.replication_lag_threshold_warning}"
+		severity              = "Warning"
+		detect_label          = "WARN"
+		disabled              = coalesce(var.replication_lag_disabled_warning, var.replication_lag_disabled, var.detectors_disabled)
+		notifications         = coalescelist(var.replication_lag_notifications_warning, var.replication_lag_notifications, var.notifications)
+		parameterized_subject = "[{{ruleSeverity}}]{{{detectorName}}} {{{ruleName}}} ({{inputs.signal.value}}) on {{{dimensions}}}"
+	}
+
+}
+
+resource "signalfx_detector" "commands" {
+	name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] AWS ElastiCache redis commands"
+
+	program_text = <<-EOF
+		A = data('GetTypeCmds', filter=filter('namespace', 'AWS/ElastiCache') and filter('stat', 'lower') and filter('CacheNodeId', '*') and ${module.filter-tags.filter_custom})${var.commands_aggregation_function}
+		B = data('SetTypeCmds', filter=filter('namespace', 'AWS/ElastiCache') and filter('stat', 'lower') and filter('CacheNodeId', '*') and ${module.filter-tags.filter_custom})${var.commands_aggregation_function}
+		signal = (A + B).${var.commands_transformation_function}(over='${var.commands_transformation_window}')
+		detect(when(signal <= ${var.commands_threshold_critical})).publish('CRIT')
+		detect(when(signal <= ${var.commands_threshold_warning})).publish('WARN')
+	EOF
+
+	rule {
```

Comment:
I would delete this one which is a duplicate for datadog "rigid" behavior which does not allow only warning